sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1767,8 +1785,48 @@ def _more_tags(self):
             }
         }
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.4
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = (
+            MetadataRouter(owner=self.__class__.__name__)
+            .add(
+                estimator=self.get_estimator(),
+                method_mapping=MethodMapping().add(callee="fit", caller="fit"),
+            )
+            .add(
+                splitter=self.cv,
+                method_mapping=MethodMapping().add(callee="split", caller="fit"),
+            )
+        )
+        return router
+
+    def get_estimator(self, **params):
+        estimator = self._get_estimator()
```

Comment:
@adrinjalali I seem to be running into a complication with this set of estimators. Basically they don't have a parameter that specifies the base estimator. Instead they each have _get_estimator() overridden which returns the corresponding model. Now this makes it difficult to set requests on these base estimators. I tried setting the fit request here but the problem with this is that if we pass split params like groups they will get added here as well. Any idea how we can approach this?

Reply:
We don't actually need to do any routing for the sub-estimator, since they're our linear models and we know they only support `sample_weight`, and our `*CV` classes take `sample_weight` explicitly in their `fit` signature and they can be seen as _consumers_ of it. So the routing mechanism would only be for the CV object and not the sub-estimator, since the sub-estimator is not exposed to the user anyway.